src/idiomatic/leveraging-the-type-system/extension-traits/trait-method-conflicts.md

```diff
@@ -0,0 +1,64 @@
+---
+minutes: 5
+---
+
+# Trait Method Conflicts
+
+What happens when you have a name conflict between two different trait methods
+implemented for the same type?
+
+```rust,compile_fail
+mod ext {
+    pub trait Ext1 {
+        fn is_palindrome(&self) -> bool;
+    }
+
+    pub trait Ext2 {
+        fn is_palindrome(&self) -> bool;
+    }
+
+    impl Ext1 for &str {
+        fn is_palindrome(&self) -> bool {
+            self.chars().eq(self.chars().rev())
+        }
+    }
+
+    impl Ext2 for &str {
+        fn is_palindrome(&self) -> bool {
+            self.chars().eq(self.chars().rev())
+        }
+    }
+}
+
+pub use ext::Ext1;
+pub use ext::Ext2;
+
+// Which method is invoked?
+// The one from `Ext1`? Or the one from `Ext2`?
+assert!("dad".is_palindrome());
+```
+
+<details>
+
+- The trait you are extending may, in a newer version, add a new trait method
+  with the same name of your extension method. Or another extension trait for
+  the same type may define a method with a name that conflicts with your own
+  extension method.
+
+  Survey the class: what do the students think will happen in the example above?
```

Comment:
I think the approach of surveying the class is something specific to the instructor and situation. I would do this without prompting for small-group, in-person instruction, but for a large or shy group, or when using Zoom or Meet, it's not a very effective strategy (it results in lots of empty air).

So, it's a very minor point and not worth changing, but I think the question in the comment is sufficient here, and does not need repeating in the notes. Instructors will prompt from the comment if appropriate, or address it in a manner appropriate to the context and their teaching style.